examples/custom-client-cxx/cli.cpp

```diff
@@ -0,0 +1,365 @@
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  try {
+    auto current = client->GetCurrent();
+
+    CheckInResult res = !is_local_update?
+                            client->CheckIn():
+                            client->CheckInLocal(&local_update_source);
+    if (res.status != CheckInResult::Status::Ok && res.status != CheckInResult::Status::OkCached) {
+      LOG_WARNING << "Unable to update latest metadata";
+      return EXIT_FAILURE;
+    }
+
+    aklite::tuf::TufTarget selected_target;
+    if (target_name.empty()) {
+      auto latest = res.GetLatest();
+      if (client->IsRollback(latest)) {
+        LOG_INFO << "Latest Target is marked for causing a rollback and won't be installed: " << latest.Name();
+      } else {
+        LOG_INFO << "Found Latest Target: " << latest.Name();
+        selected_target = latest;
+      }
+    } else {
+      for (auto const &target: res.Targets()) {
+        if (target.Name() == target_name) {
+          selected_target = target;
+          break;
+        }
+        LOG_ERROR << "Target " << target_name << " not found or is incomplete";
+        return EXIT_FAILURE;
+      }
+    }
+
+    if (client->IsRollback(selected_target) && target_name.empty() && current.Name() == selected_target.Name()) {
+      // Handle the case when Apps failed to start on boot just after an update.
+      // This is only possible with `pacman.create_containers_before_reboot = 0`.
+      LOG_INFO << "The currently booted Target is a failing Target, finding Target to rollback to...";
+      const TufTarget rollback_target = client->GetRollbackTarget();
+      if (rollback_target.IsUnknown()) {
+        LOG_ERROR << "Failed to find Target to rollback to after failure to start Apps at boot of a new sysroot";
+        return EXIT_FAILURE;
+      }
+      selected_target = rollback_target;
+      LOG_INFO << "Rollback Target is " << selected_target.Name();
+    }
+
+    if (selected_target.Name() != current.Name() && !client->IsRollback(selected_target)) {
+      std::string reason = "Updating from " + current.Name() + " to " + selected_target.Name();
+      auto installer = client->Installer(selected_target, reason, "", InstallMode::All, &local_update_source);
+      if (!installer) {
+        LOG_ERROR << "Failed to retrieve target " << selected_target.Name() << " metadata from DB, skipping update";
+        return EXIT_FAILURE;
+      }
+      auto dres = installer->Download();
+      if (dres.status != DownloadResult::Status::Ok) {
+        LOG_ERROR << "Unable to download target: " << dres;
+        return EXIT_FAILURE;
+      }
+    }
+    LOG_INFO << "Nothing to do";
+    return EXIT_SUCCESS;
+
+  } catch (const std::exception &exc) {
+    LOG_ERROR << "Failed to find or update Target: " << exc.what();
+    return EXIT_FAILURE;
+  }
+}
+
+int cmd_install(std::string target_name, std::string local_repo_path) {
```

Comment:
The install command has the same two issues as the `pull` command.

Reply:
Fixed